multy_core/src/bitcoin/bitcoin_transaction.cpp

```diff
@@ -709,15 +701,43 @@ void BitcoinTransaction::update()
         //
         // Not adding change for leftovers less than said limits increases fee for this TX.
 
-        const int64_t avarage_input_size = is_segwit() ? 100 : 150;
+        const int64_t average_input_size = is_segwit() ? 100 : 150;
         const int64_t min_tx_cost_per_byte = 1;
-        if (remainder > avarage_input_size * min_tx_cost_per_byte)
+        if (remainder > average_input_size * min_tx_cost_per_byte)
         {
             // not setting value with set_value(), since that would fail for read-only property.
             change_destination->amount.get_value() += remainder;
+
+            tx_size = get_transaction_serialized_size(WITH_POSITIVE_CHANGE_AMOUNT);
+            expected_fee = tx_size * *m_fee->amount_per_byte;
+            current_fee = calculate_diff();
+            while (expected_fee > current_fee)
+            {
+                // get shortage fee from change_destination amount
+                const BigInt shortage_fee = expected_fee - current_fee;
```

Comment:
Please use same pattern as above, that is
```
// Note that diff is always negative here.
diff current_fee - expected_fee
change_destination->amount.get_value() += diff;
```

That would simplify code and improve maintenance a bit.